simulator/simple_planning_simulator/test/test_simple_planning_simulator.cpp

```diff
@@ -135,32 +99,13 @@ void resetInitialpose(rclcpp::Node::SharedPtr sim_node, std::shared_ptr<PubSubNo
 }
 
 void sendGear(
-  uint8_t gear, rclcpp::Node::SharedPtr sim_node,
-  std::shared_ptr<PubSubNode> pub_sub_node)
+  uint8_t gear, rclcpp::Node::SharedPtr sim_node, std::shared_ptr<PubSubNode> pub_sub_node)
 {
   GearCommand cmd;
   cmd.stamp = sim_node->now();
-  cmd.gear = gear;
+  cmd.command = gear;
   for (int i = 0; i < 10; ++i) {
```

Comment:
Looping a fixed number of times will increase test time and decrease determinism. It's better to get the state from Node's public API if possible.
(Same as L93, L125)

Reply:
@KeisukeShima  Thank you for your comment! For now, such an API is not supported in the `simple_planning_simulator`. Should we create it for this kind of test? Or do you know any general API in the `rclcpp::Node` or somewhere?

Reply:
@TakaHoribe Thank you for the valuable information!

> Should we create it for this kind of test? Or do you know any general API in the rclcpp::Node or somewhere?

If you want to monitor the initialization state, there is a way to manage the Node state with `LifecycleNode`.
However, I don't think it is necessary to create an API just for testing.

Sending the initialization message 10 times seems strange, but I understood that it is a workaround.
If the test takes a long time, you can also use simulated time using `use_sim_time`. Contact me if you are interested. :smile:

Reply:
@KeisukeShima I feet the `10` number seems strange as well. We are willing to replace this test with `launch_testing`, so let us discuss the test design with you then!